cpp/include/cugraph/algorithms.hpp

```diff
@@ -1380,6 +1384,37 @@ extract_ego(raft::handle_t const& handle,
             vertex_t n_subgraphs,
             vertex_t radius);
 
+/**
+ * @brief returns induced EgoNet subgraph(s) of neighbors centered at nodes in source_vertex within
+ * a given radius.
```

Comment:
Isn't this identical to k-hop neighbors?

Reply:
Umm... maybe not... I assume identical only when k=1.

Reply:
Essentially the ego net for a vertex v1 and a radius of k is the following:
* compute the distance of every vertex in the graph from v1
* induce subgraph for the subset of vertices that are of distance <= k

So for radius = 1, that will include v1, all of its neighbors, the edges to its neighbors and the edges between its neighbors.
For radius = 2, that will include v1, all of its neighbors, all of its 2-hop neighbors, the edges to the 1-hop neighbors, the edges between the 1-hop neighbors, the edges from the 1-hop neighbors to the 2-hop neighbors and the edges between the 2-hop neighbors.
etc.

Reply:
Gotcha, yeah... so this is basically finding [0, radius]-hop neighbors then running induced subgraph, so this can be a bit different from just running radius-hop neighbors then running induced subgraph.